test-network-function/networking/suite.go

```diff
@@ -440,17 +442,23 @@ func testListenAndDeclared(env *config.TestEnvironment) {
 
 			err = listeningPortList(commandlisten, nodeOc.Context, listeningPorts)
 			if err != nil {
-				log.Errorf("Failed: There are no listening ports for pod name %s in pod namespace %s", podUnderTest.Name, podUnderTest.Namespace)
+				log.Errorf("Failed to get listening ports for pod name %s in pod namespace %s due to %v", podUnderTest.Name, podUnderTest.Namespace, err)
 				continue
 			}
-			// compare between declaredPort,listeningPort and return the common.
-			res, undeclaredPorts = checkIfListenIsDeclared(listeningPorts, declaredPorts)
-			if !res {
-				tnf.ClaimFilePrintf("TC failed : port is listening but not declared in pod name %s and pod namespace is %s", podUnderTest.Name, podUnderTest.Namespace)
-			}
+			// compare between declaredPort,listeningPort
+			undeclaredPorts = checkIfListenIsDeclared(listeningPorts, declaredPorts)
 			for k := range undeclaredPorts {
 				tnf.ClaimFilePrintf("The port %d on protocol %s in pod name %s and pod namespace is %s not declared.", k.port, k.protocol, podUnderTest.Name, podUnderTest.Namespace)
 			}
+			if len(undeclaredPorts) != 0 {
+				failedPods[podUnderTest.Name] = podUnderTest.Namespace
+			}
+		}
+		for k := range failedPods {
```

Comment:
Fail only works once. No need to use a for loop unless you want to build a Fail message with the content from the pod list.